### PR TITLE
More tests for new feature try

### PR DESCRIPTION
**Pull request by thibaultduponchelle**
Hello,

This is some more tests for the try/catch feature 😀 

I tried to stress a bit the feature with some (kind of) real use cases (signal, eval, propagation).

### CODE REVIEW

**Comment by thibaultduponchelle** [comment]
@leonerd 😃

**Comment by jkeenan** [comment]
> Hello,
> 
> This is some more tests for the try/catch feature grinning
> 
> I tried to stress a bit the feature with some (kind of) real use cases (signal, eval, propagation).

@leonerd, can you review this p.r.?  I'm inclined to apply it within 7 days unless you indicate otherwise.

File: t/op/try.t [hunk 1]
```diff
@@ -74,6 +74,88 @@ no warnings 'experimental::try';
     is(f(), "return inside try", 'return inside try');
 }
 
+no warnings "exiting";
```
Comment (leonerd) [inline t/op/try.t]: Move this inside the scope brackets of the test it applies to

File: t/op/try.t [hunk 2]
```diff
@@ -74,6 +74,88 @@ no warnings 'experimental::try';
...
+        try {
+            $result .= "d";
+            eval "last";
+            $result .= "e";
+        } catch ($e) {
+            $result .= "f";
+        }
+    }
+
+    is($result, "d", 'eval last block outside try');
```
Comment (leonerd) [inline t/op/try.t]: ITYM "inside try"

**Review by leonerd — CHANGES_REQUESTED** [review]
Needs a couple of small changes but otherwise looks good.

File: t/op/try.t (hunk 2 above)
Comment (haarg) [reply t/op/try.t]: The block that the last is terminating is outside the try.

**Comment by jkeenan** [comment]
@thibaultduponchelle , could you update your pull request based on the feedback ^^ from @leonerd and @haarg ?  Thanks.

File: t/op/try.t (hunk 1 above)
Comment (thibaultduponchelle) [reply t/op/try.t]: done

File: t/op/try.t (hunk 2 above)
Comment (thibaultduponchelle) [reply t/op/try.t]: I edited the descriptions to make them clearer

**Comment by thibaultduponchelle** [comment]
Thank you @leonerd and @haarg for the review (and thank you @jkeenan). I updated according to the remarks 👍🏼

**Comment by jkeenan** [comment]
> Thank you @leonerd and @haarg for the review (and thank you @jkeenan). I updated according to the remarks 👍🏼

@leonerd and @haarg, can you review the contributor's most recent changes in this pull request?  Thanks.

**Comment by leonerd** [comment]
Overall no huge objection, though in parts I'm unsure quite the value being added by the tests. While "more tests are always good", there gets to a point where we're just testing many minor variations on the same idea, which of course adds size and time.

The signal one at the end, in particular, doesn't actually test signal propagation *through* a `try` boundary, only entirely within on.

**Comment by jkeenan** [comment]
> 
> The signal one at the end, in particular, doesn't actually test signal propagation _through_ a `try` boundary, only entirely within on.

@thibaultduponchelle, can you respond to this point raised by @leonerd ?

Thank you very much.
Jim Keenan

**Comment by thibaultduponchelle** [comment]
@jkeenan I’m going to do the change 👍🏼

**Comment by thibaultduponchelle** [comment]
@jkeenan @leonerd I moved SIGINT handler out of `try {}` block and added an outer and inner block to make it clearer that correct `catch` is reached.

**Pull request closed without merging**